util/rate_limiter.cc

```diff
@@ -136,96 +136,85 @@ void GenericRateLimiter::Request(int64_t bytes, const Env::IOPriority pri,
   Req r(bytes, &request_mutex_);
   queue_[pri].push_back(&r);
 
+  // A thread representing a queued request coordinates with other such threads.
+  // There are two main duties.
+  //
+  // (1) Waiting for the next refill time.
+  // (2) Refilling the bytes and granting requests.
```

Comment:
Very useful comment about the heavy-working duty, regardless of whether we want to keep the concept of leader or not in our impl. I integrated these two comment lines in the readability PR because it helps improve the readability without any big code changes, aligning with the readability PR []. I credited you in here [], but do let me know this credit does not change the fact that I was a thief of good comments!

Reply:
Yes please take whatever's useful